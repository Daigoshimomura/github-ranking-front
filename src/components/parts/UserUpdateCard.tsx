import React, { MouseEventHandler } from "react";
import { Card, CardContent, Typography, Button, Box } from "@material-ui/core";
import UpdateIcon from "@material-ui/icons/Update";

interface UserUpdateCardProps {
  lastupdateDate: Date;
  onClickUpdateButton: MouseEventHandler;
}

export class UserUpdateCard extends React.Component<UserUpdateCardProps, {}> {
  constructor(props: UserUpdateCardProps) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }
  onClick(event: React.MouseEvent<Element, MouseEvent>) {
    this.props.onClickUpdateButton(event);
  }
  render() {
    return (
      <Card style={{ maxWidth: 270, display: "flex" }}>
        <CardContent>
          <Button
            variant="contained"
            color="primary"
            endIcon={<UpdateIcon>update</UpdateIcon>}
            onClick={event => {
              this.props.onClickUpdateButton(event);
            }}
          >
            update
          </Button>
          <Box mt={3.5}>
            <Typography color="textSecondary">
              Last update :{this.props.lastupdateDate.toDateString()}
            </Typography>
          </Box>
        </CardContent>
      </Card>
    );
  }
}
