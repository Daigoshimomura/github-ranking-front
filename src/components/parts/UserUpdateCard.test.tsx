import { mount } from "enzyme";
import React from "react";
import { UserUpdateCard } from "./UserUpdateCard";
import { Card, Button } from "@material-ui/core";
import renderer from "react-test-renderer";

const date = new Date();
const testMock = jest.fn();

describe("UserUpdateCard", () => {
  it("snapshot", () => {
    const tree = renderer
      .create(
        <UserUpdateCard lastupdateDate={date} onClickUpdateButton={testMock} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("レンダリングの確認", () => {
    const wrapper = mount(
      <UserUpdateCard lastupdateDate={date} onClickUpdateButton={testMock} />
    );
    expect(wrapper.find(Card).length).toBe(1);
  });
  it("propsをちゃんと渡せてることの確認", () => {
    const wrapper = mount(
      <UserUpdateCard lastupdateDate={date} onClickUpdateButton={testMock} />
    );
    expect(wrapper.prop("lastupdateDate")).toBe(date);
    expect(wrapper.prop("onClickUpdateButton")).toBe(testMock);
  });
  it("クリック時にコールバックが実行されることを確認", () => {
    const wrapper = mount(
      <UserUpdateCard lastupdateDate={date} onClickUpdateButton={testMock} />
    );
    wrapper.find(Button).simulate("click");
    expect(testMock).toHaveBeenCalled();
  });
});
