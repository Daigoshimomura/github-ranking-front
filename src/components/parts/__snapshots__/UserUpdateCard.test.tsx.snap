// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserUpdateCard snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  style={
    Object {
      "display": "flex",
      "maxWidth": 270,
    }
  }
>
  <div
    className="MuiCardContent-root"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        update
        <span
          className="MuiButton-endIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 10.12h-6.78l2.74-2.82c-2.73-2.7-7.15-2.8-9.88-.1-2.73 2.71-2.73 7.08 0 9.79s7.15 2.71 9.88 0C18.32 15.65 19 14.08 19 12.1h2c0 1.98-.88 4.55-2.64 6.29-3.51 3.48-9.21 3.48-12.72 0-3.5-3.47-3.53-9.11-.02-12.58s9.14-3.47 12.65 0L21 3v7.12zM12.5 8v4.25l3.5 2.08-.72 1.21L11 13V8h1.5z"
            />
          </svg>
        </span>
      </span>
    </button>
    <div
      className="MuiBox-root MuiBox-root-73"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        Last update :
        Mon Feb 24 2020
      </p>
    </div>
  </div>
</div>
`;
